### ISSUE TITLE
V-Rally 2 doesn't recognize songs, plays only first one

### ISSUE THREAD

**Issue by leipero** [ayuanx/ogg-winmm#6]
I've tried your last build, V-Rally 2 doesn't play songs at all, track number is grayed out in audio section, however, once you "select" the grayed out section, first track is played, you can't change it. If you enter the race, first track will play as well, but as soon as you exit and start new race, it should be the next track playing (track 02), but that doesn't happen.

It happens in all forks of ogg-winmm, even the latest build as well. It seems to be an player issue.

**Comment by leipero** [comment]
With CD it works, please notify me to remove the link, I share my custom made .iso so you can test it.
https://drive.google.com/file/d/1EO5OI1Hq3wVi6G-HwTjlkPYoproe9VCL/view?usp=sharing

**Comment by leipero** [comment]
Did not have game installed, just installed it now, and realized that after fixed version (ogg-winmm release v2022.02.28), the very next one (ogg-winmm release v2022.05.26) and every version since does not work with V-Rally 2 EE anymore.

Just letting you know for the betterment of the software. Maybe MCI_INFO requires removed monitor thread?

**Comment by ayuanx** [comment]
No, I don't think that is the reason.
Please try this debug version and upload the log.

[winmm.zip](https://github.com/ayuanx/ogg-winmm/files/10931149/winmm.zip)

**Comment by leipero** [comment]
Ok, sorry for the late reply, here are two logs:
1. Just entering game, selecting first available mode and playing.
2. Doing the same as 1 plus after leaving the race going to options, audio, audio track (grayed out).

Both cases, no music, track 1 grayed out unable to change.
[winmm_1.log](https://github.com/ayuanx/ogg-winmm/files/10939184/winmm_1.log)
[winmm_2.log](https://github.com/ayuanx/ogg-winmm/files/10939186/winmm_2.log)

**Comment by ayuanx** [comment]
Your log shows there is no track available.
```
ogg-winmm music directory is C:\Program Files (x86)\V-Rally 2 Expert Edition\MUSIC
ogg-winmm searching tracks...
Emulating total of 0 CD tracks.
```
Did you forget to rip the soundtracks to ogg files? And the file names can NOT contain whitespaces.

**Comment by leipero** [comment]
> Your log shows there is no track available.
> 
> ```
> ogg-winmm music directory is C:\Program Files (x86)\V-Rally 2 Expert Edition\MUSIC
> ogg-winmm searching tracks...
> Emulating total of 0 CD tracks.
> ```
> 
> Did you forget to rip the soundtracks to ogg files? And the file names can NOT contain whitespaces.

Yes, I saw that, simply changing version of ogg-winmm to release v2022.02.28 makes it work.
Here is the whole folder so you can test, just let me know to remove the link once downloaded:
https://gofile.io/d/DhDSHv

**Comment by ayuanx** [comment]
The problem is the missing of Track02.ogg.

It expects the music track on CD to start from either Track01 or Track02.
If both are missing, it stops searching.

You can easily fix this by adding back Track02.ogg.

**Comment by leipero** [comment]
Oh, ok, I didn't try that, but it did occur to me at some point. The main reason why I named it starting Track03 is because it didn't work on previous versions of ogg-winmm if I start with Track02 as suggested, and if I remember correctly (but I can't be sure), it was also the issue of missing tracks if not starting with 03, so last track would be missing (only 8 tracks out of 9), but don't quote me on that, it could be I misremember it.

I don't remember to be honest, I think it was standard CD, data on Track01, audio for Track02+.

I assume you tried it, but I will try it latter, and edit post with the results, thanks, that's an easy fix indeed, I should have tried it.
